File: interpreter/typescript/team-lead-expression.ts
import {IJobExpression} from "./job-expression.interface";
import {Context} from "./context";

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
export class TeamLeadExpression implements IJobExpression {
  public interpret(context: Context): void {
    if(context.formula.includes('L')) {
      context.totalPoint += 7000;
    }
  }
}

File: interpreter/go/team-lead-expression.go
package main

import "strings"

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
type teamLeadExpression struct{}

func NewTeamLeadExpression() *teamLeadExpression {
	return &teamLeadExpression{}
}

func (e *teamLeadExpression) Interpret(ctx *Context) {
	if strings.Contains(ctx.Formula, "L") {
		ctx.TotalPoint += 7000
	}
}
